modules/core/src/main/java/org/apache/synapse/endpoints/WSDLEndpoint.java

```diff
@@ -93,15 +105,27 @@ public String getServiceName() {
         return serviceName;
     }
 
+    public String getGetOriginalWsdlServiceName() {
+        return this.getOriginalWsdlServiceName;
+    }
+
     public void setServiceName(String serviceName) {
-        this.serviceName = serviceName;
+        this.getOriginalWsdlServiceName = serviceName;
+        String fetchedServiceName = ConfigUtils.fetchEnvironmentVariables(serviceName);
+        this.serviceName = fetchedServiceName;
     }
 
     public String getPortName() {
         return portName;
     }
 
+    public String getOriginalWsdlPort() {
+        return this.originalWsdlPort;
+    }
+
     public void setPortName(String portName) {
-        this.portName = portName;
+        this.originalWsdlPort = portName;
+        String fetchedPortName = ConfigUtils.fetchEnvironmentVariables(portName);
```

Comment:
same comment as AddressEndpoint. fetchedPortName local variable is not required.

Reply:
Resolved https://github.com/apache/synapse/pull/55/commits/61fb9e4da909b00d6da8dfe9fb57a3c0a5892793